R/proxy.R

```diff
@@ -207,6 +207,10 @@ vec_data <- function(x) {
   vec_assert(x)
   x <- vec_proxy(x)
 
+  if (is.data.frame(x)) {
+    return(new_data_frame(x, row.names = attr(x, "row.names")))
```

Comment:
Does this have performance implications? Doesn't this materialize compact row names?

Reply:
oops you're right, `attr()` doesn't return a character vector unlike `row.names()`, but it does materialise. We should probably use `.row_names_info(x, 0)` here.

Reply:
we have used `.row_names_info(data, type = 0L)` many times in dplyr. is it worth making an rlang helper for that?

Reply:
Now done, thanks!